app/docker/views/containers/console/exec.html

```diff
@@ -51,16 +51,22 @@
             </div>
             <div class="form-group">
               <div class="col-sm-12">
-                <button type="button" class="btn btn-primary" ng-disabled="state.connected" button-spinner="state.connected" ng-click="connect()">
-                  <span ng-hide="state.leaveNetworkInProgress">Connect</span>
-                  <span ng-show="state.leaveNetworkInProgress">Connecting...</span>
+                <button type="button" class="btn btn-primary" ng-disabled="state.connecting || state.connected || !container.State.Running" ng-click="connectExec()">
+                  <span>Connect</span>
                 </button>
+                <span class="small text-danger" ng-hide="container.State.Running">
+                  <i class="fa fa-exclamation-triangle" aria-hidden="true"></i>
+                  The container is not running.
+                </span>
               </div>
             </div>
           </div>
-          <div ng-if="state.connected">
+          <div ng-if="state >= states.connecting">
```

Comment:
Not easy to read without knowing the states and how they are ordered.
I'd prefer using `ng-if="state !== states.disconnected"` here.

Reply:
Done